watchexec/watchexec
# Add option to exit when stdin ends

This is a first draft/approach to the `--stdin-quit` functionality raised in #440. I am not experienced with the code base and I'm not very experienced with Rust. I'm happy to change any and all of this. I just hope it isn't so wrong as to be completely useless.

Important to note that this first draft always quits on stdin ending regardless of the flag usage. I couldn't see how the runtime config is used to influence the behaviour as it is not clearly used in either the event generation code that I inspected or the `on_action` code but I assume I'm missing something. It would be reasonably easy to pass to either to ensure either the `Keyboard::Eof` tagged event isn't generated without the flag or that the `Stop/Exit` outcome isn't generated for the event without the flag.

### Description

The watchexec cli gains a '--stdin-quit' flag which will cause the watchexec to end & exit when stdin for watchexec closes. This is to make it easier to use watchexec as a subprocess and make sure that it ends with the parent process ends, provided the parent process controls the stdin to the watchexec process.

We model the end of stdin as a keyboard event as it is possible to send 'eof' from a terminal using 'ctrl-d'. We keep reading lines from stdin and if we get back 0 bytes we assume that means that the stream has closed. This is a technique I have used elsewhere but I'm not that experienced so there might be a much better way to do it.

### Manual testing

A simple cargo run with ctrl-d to check that stdin closing ends the watching. I'm not sure if there are better ways.

```
$ cargo run -q -- "echo hello"
hello
hello
$ # Process has ended due to ctrl-d input
```

### Checklist

- [x] Ignore Eof event when `--stdin-quit` is not set
- [ ] Add unit tests as appropriate
- [x] Improve naming of field and functions
- [x] Add documentation to fix clippy warnings
- [x] Add man page entry
